### PR TITLE
Use virtualenv>=20

### LINKED ISSUE #231: incompatibility with virtualenv 20
`virtualenv` version 20 waas released three days ago. Since then, using venv-update unmodified results in this error:

```
OSError: [Errno 39] Directory not empty: 
   '$HOME/.cache/venv-update/3.1.1/src.tmp' -> 
   '$HOME/.cache/venv-update/3.1.1/src'
```

This is due to a check for existance of virtualenv.py always failing, which is in turn due to virtualenv project refactoring from a module to a package with virtualenv/__main__.py. I tried to simply patch venv-update to handle the new structure, but the changes are too extensive.

As a short-term fix, this works for me (tm):

```!diff
--- sbin/venv-update.orig       2020-02-13 11:25:28.210285813 -0800
+++ sbin/venv-update    2020-02-13 11:33:31.696390868 -0800
@@ -200,7 +200,7 @@
         scratch_python = venv_python(scratch.venv)
         # TODO: do we allow user-defined override of which version of virtualenv to install?
         tmp = scratch.src + '.tmp'
-        run((scratch_python, '-m', 'pip.__main__', 'install', 'virtualenv', '--target', tmp))
+        run((scratch_python, '-m', 'pip.__main__', 'install', 'virtualenv<20', '--target', tmp))
 
         from os import rename
         rename(tmp, scratch.src)
```

### PR DESCRIPTION
**Pull request by kaisen**
virtualenv>=20 contains a big refactor of the inner workings of virtualenv which broke venv-update. This PR accounts for the major changes and requires virtualenv>=20 moving forward.

Notable changes:
* virtualenv>=20 is required
* The old `adjust_options` hook doesn't exist anymore in virtualenv, so the logic to check if an old virtualenv is valid needs to be done before virtualenv is invoked
* Checking if `--system-site-packages` was passed to virtualenv is now done by checking for a config in the new pyenv.cfg file
* Major version bump to 4.0.0

Fixes #231.

### CODE REVIEW

File: venv_update.py [hunk 1]
```diff
@@ -235,25 +234,13 @@ def get_original_path(venv_path):  # TODO-TEST: a unit test
...
-        os.path.join(os.path.dirname(site.__file__), 'no-global-site-packages.txt')
-    ) else
-    1
-)"""
-    ))
-    system_site_packages = int(system_site_packages)
-    assert system_site_packages in (0, 1)
-    return bool(system_site_packages)
+def has_system_site_packages(venv_path):
+    from pathlib import Path
```
Comment (asottile) [inline venv_update.py:238]: this breaks python 2

File: venv_update.py [hunk 2]
```diff
@@ -235,25 +234,13 @@ def get_original_path(venv_path):  # TODO-TEST: a unit test
...
+    from virtualenv.create.pyenv_cfg import PyEnvCfg
+    pyenv_cfg = PyEnvCfg.from_folder(Path(venv_path))
+    include_system_site_packages = pyenv_cfg['include-system-site-packages']
+    assert include_system_site_packages in ('true', 'false')
+    return True if include_system_site_packages == 'true' else False
```
Comment (asottile) [inline venv_update.py:243]: [`True if ... else False` is better written as just `...`](https://github.com/asottile/scratch/wiki/PythonHintsForNoobs#useless-branches)

File: venv_update.py [hunk 3]
```diff
@@ -235,25 +234,13 @@ def get_original_path(venv_path):  # TODO-TEST: a unit test
...
-)"""
-    ))
-    system_site_packages = int(system_site_packages)
-    assert system_site_packages in (0, 1)
-    return bool(system_site_packages)
+def has_system_site_packages(venv_path):
+    from pathlib import Path
+    from virtualenv.create.pyenv_cfg import PyEnvCfg
+    pyenv_cfg = PyEnvCfg.from_folder(Path(venv_path))
+    include_system_site_packages = pyenv_cfg['include-system-site-packages']
```
Comment (asottile) [inline venv_update.py:241]: I would recommend using `configparser` directly -- there's no benefits from using the `virtualenv` one here

File: CI/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
 # minimal set of packages for github actions to run our test suites
 codecov
 tox==3.21.2
-virtualenv<20
+virtualenv>=20
```
Comment (asottile) [inline CI/requirements.txt:4]: I would recommend a sufficiently newer version of virtualenv 20.x -- the early versions were quite buggy.  the minimum bound in `pre-commit` is a reasonable one: https://github.com/pre-commit/pre-commit/blob/24d9dc7469f9b752b5c772d5cc829d95b43e0e8c/setup.cfg#L31

**Review by asottile — COMMENTED** [review]
the invalid check can be greatly simplified as virtualenv 20.x doesn't do major copies of things

here's some prior art: https://github.com/pre-commit/pre-commit/blob/24d9dc7469f9b752b5c772d5cc829d95b43e0e8c/pre_commit/languages/python.py#L166-L186

**Review by analogue — APPROVED** [review]

**Review by ymilki — APPROVED** [review]
Thanks for all the attention to get all the tests passing!

**Pull request merged**